Implement << operator for std basic_ostream and etl string_view

mike919192 opened the pull request:
I recently wanted to print the contents of etl::string_view to std::cout and I realized that there is not yet an overload for the `<<` operator.  I decided to experiment with adding the function overload and all seems to be working.

I put the function overload and unit test behind `ETL_USING_STL` macro.  It will not be including in no stl configs.  For the unit test I am using std::stringstream variants, but I have also tested with std::cout.

If this is acceptable I would think that the same could be done for etl::string.  Let me know of any comments.  Thanks

jwellbelove commented:
It would probably be better to create overloads of `std::ostream <<` operator for `etl::basic_string` and `etl::basic_string_view`.

mike919192 commented:
Sorry I'm not sure what you mean by the comment.  `std::ostream` is a typedef for `std::basic_ostream<char>`.  I implemented the template function so that it can be used with `char` / `wchar_t` / etc.  That way the function can be used with whatever char type the stream supports.

The definition for the function I added:
```
template <typename T>
std::basic_ostream<T, std::char_traits<T> > &operator<<(std::basic_ostream<T, std::char_traits<T> > &os, 
                                                        etl::basic_string_view<T, etl::char_traits<T> > text)
```

jwellbelove commented:
Sorry, my mistake.

jwellbelove commented:
Add the overload for `etl::ibasic_string<T>` and I'll merge the pull request.

mike919192 commented:
> Add the overload for `etl::ibasic_string<T>` and I'll merge the pull request.

It's implemented and I think I have all the tests covered.  Let me know if anything is missing.
